### PR TITLE
Add tfa enforce flow

### LINKED ISSUE #5682: "Enforce 2FA" doesn't trigger on login
1) The issue and what you expected to happen

When logging in with a user in a role that has enforce 2fa enabled, the app should force you into the tfa-setup flow immediately.

2) Exact steps to reproduce this issue

Enable enforce 2fa, and login

### LINKED ISSUE #9883: Update right for directus_users.tfa_secret missing at minimal app permissions
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues?q=is%3Aissue) that already describes my problem.

### Describe the Bug

Users with a role assigned with only the minimal or recommended system permissions are unable to activate 2FA since the update right on their tfa_secret field is missing.

~~I have a PR ready.~~

### To Reproduce

1. Create a new role with only minimal or recommended system permissions.
2. Create a new user with this role.
3. Enable 2FA for this user.
4. Scan the QR code.
5. Enter OTP

### Errors Shown

You don't have permission to access this.

### What version of Directus are you using?

9.0.1

### What version of Node.js are you using?

v16.13.0

### What database are you using?

MSSQL

### What browser are you using?

Chrome

### What operating system are you using?

Windows

### How are you deploying Directus?

running locally

### PR DESCRIPTION
**Pull request by Nitwel**
closes #5682

https://user-images.githubusercontent.com/22577866/132015695-05db5ba2-47ef-4ab1-b188-9037514ea563.mp4

### CODE REVIEW

**Review by joselcvarela — APPROVED** [review]

**Comment by licitdev** [comment]
## Demo

https://user-images.githubusercontent.com/26413686/170491429-48fe1cf0-99f1-43e8-a1a4-498c55f1b9f8.mov

## Admin disabling user's 2FA (For resetting when required)

https://user-images.githubusercontent.com/26413686/170491573-0a5234e0-a1b4-45d6-a140-db28ec9e3828.mov

**Review by azrikahar — APPROVED** [review]

File: app/src/modules/settings/routes/roles/app-permissions.ts [hunk 1]
```diff
@@ -121,6 +121,7 @@ export const appRecommendedPermissions: Partial<Permission>[] = [
 			'avatar',
 			'language',
 			'theme',
+			'tfa_secret',
```
Comment (azrikahar) [inline app/src/modules/settings/routes/roles/app-permissions.ts:124]: This does resolve #9883 for new setups, but existing setups will already have the recommended permissions _without_ this field, thus getting error 403 when entering their OTP in the enforce tfa flow.

Perhaps we'll need to add a notice/warn about this 🤔

Comment (rijkvanzanten) [reply app/src/modules/settings/routes/roles/app-permissions.ts:124]: I think the API endpoint to enable/disable TFA shouldn't rely on regular user permissions. It should require the user's ID, current user's password, and then allow updating tfa_secret without relying on the role permissions.

Comment (rijkvanzanten) [reply app/src/modules/settings/routes/roles/app-permissions.ts:124]: Hmm that's also true, i'm just really worried about the annoyance / confusion that is requiring the explicit update permission to this one field. We'd also don't want to expose updating that particular field for _other_ users, so if you're not paying close attention to the rules configured for the role, you end up allowing any user to set the tfa secret for other users 🤔 I think TFA configuration might be an exception to our regular permissions handling, as it's such a security sensitive topic

Comment (licitdev) [reply app/src/modules/settings/routes/roles/app-permissions.ts:124]: Awesome, will look into patching this up later. 👍🏻

Comment (AustinPhillipTaylor) [reply app/src/modules/settings/routes/roles/app-permissions.ts:124]: @rijkvanzanten No worries 😂 I completely understand these are two different problems with very different requirements.

Comment (licitdev) [reply app/src/modules/settings/routes/roles/app-permissions.ts:124]: Just noticed that it was already mentioned in the docs https://github.com/directus/directus/pull/7805/commits/5a129b6815cc4ff1fc3319cec874303f6f37dd8e 🙈  
Note to revert the docs change after the patch.

Comment (licitdev) [reply app/src/modules/settings/routes/roles/app-permissions.ts:124]: Done.

**Review by br41nslug — APPROVED** [review]

**Pull request merged**